samples/server/petstore/go-chi-server/go/model_gender.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * OpenAPI Petstore
+ *
+ * This is a sample server Petstore server. For this sample, you can use the api key `special-key` to test the authorization filters.
+ *
+ * API version: 1.0.0
+ * Generated by: OpenAPI Generator (https://openapi-generator.tech)
+ */
+
+package petstoreserver
+
+
+import (
+	"fmt"
+	"golang.org/x/exp/maps"
+)
+
+
+
+type Gender string
+
+// List of Gender
+const (
+	MALE Gender = "male"
+	FEMALE Gender = "female"
+)
+
+// All allowed values of Gender enum
+var AllowedGenderEnumValues = map[Gender][]struct{}{
+	"male": {},
+	"female": {},
+}
+
+// IsValid return true if the value is valid for the enum, false otherwise
+func (v Gender) IsValid() bool {
+	_, ok := AllowedGenderEnumValues[v]
+	return ok
+}
+
+// NewGenderFromValue returns a pointer to a valid Gender
+// for the value passed as argument, or an error if the value passed is not allowed by the enum
+func NewGenderFromValue(v string) (Gender, error) {
+	ev := Gender(v)
+	if ev.IsValid() {
+		return ev, nil
+	} else {
+		return "", fmt.Errorf("invalid value '%v' for Gender: valid values are %v", v, maps.Keys(AllowedGenderEnumValues))
```

Comment:
could we store `maps.Keys(AllowedGenderEnumValues)` as a var?

Reply:
I'll do a privately scoped map to do validation checks on inputs and a publicly scoped slice of the enums. The map doesn't seem to be useful from a public usage of the model but the slice of valid enums did get used in my companies code base.